script.js

```diff
@@ -1,4 +1,100 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+let numOfDice = 0;
+let gameMode = "waiting for number of players";
+let playerTurn = 0;
+let arrayOfPlayers = [];
+
+//store Player's object to be used for sorting diceRolls and leaderboard
+const storePlayerObjects = (userInput) => {
+  for (let i = 0; i < userInput; i += 1) {
+    arrayOfPlayers.push({ id: i + 1, diceRolls: [] });
+  }
+};
+
+//get winner at the end of game , display leaderboard, restart the game
+const findWinner = () => {
+  let sortedScore = arrayOfPlayers.sort((a, b) => b.diceRolls - a.diceRolls);
+  let winner = sortedScore[0].id;
+  let finalMessage = `Winner is player ${winner}.\<br\><br\>\ Leaderboard\<br\>\ ${leaderBoard(
+    arrayOfPlayers
+  )}\<br\><br\>\Type number of players to replay the game.`;
+  resetGame();
+  return finalMessage;
+};
+
+//prep leaderboard
+const leaderBoard = (anArray) =>
+  anArray.reduce(
+    (acc, curVal) =>
+      `${acc} Player ${curVal.id}. Dicerolls: ${curVal.diceRolls}\<br\>\ `,
+    ""
+  );
+
+//resetGame
+const resetGame = () => {
+  numOfDice = 0;
+  gameMode = "waiting for number of players";
+  playerTurn = 0;
+  arrayOfPlayers = [];
+};
+
+//rolling dice number with Math
+const getDiceRoll = () => {
+  return 1 + Math.floor(Math.random() * 6);
+};
+//pushing the sorted diceRoll to diceArray
+const diceRollArray = () => {
+  let diceArray = [];
+  for (let n = 0; n < Number(numOfDice); n++) {
+    let diceNumber = getDiceRoll();
+    diceArray.push(diceNumber);
+    sortDiceArray(diceArray);
+  }
+  arrayOfPlayers[playerTurn].diceRolls = Number(diceArray.join(""));
+  return diceArray;
+};
+// sorting array
+const sortDiceArray = (anArray) => {
+  anArray.sort(function (a, b) {
+    return b - a;
+  });
+};
+
+//normalMode
+const normalMode = () => {
+  diceRollArray();
```

Comment:
Not sure if you experienced trouble tracking certain elements, usually a function like diceRollArray() if affecting the global variable `arrayOfPlayers` will hinder your debugging process since it's not returning anything in this function 

It would be good to have decomposed this further

Reply:
Referring to the `diceRollArray()` function